models/multiphase/d2q9_pf_cm/Dynamics.c.Rt

```diff
@@ -0,0 +1,1488 @@
+	f_in[5] = temp[5] - temp[6] + temp[7] - temp[8];
+	//central moments from raw moments
+	temp[3] = f_in[0]*ux2 - 2.0*f_in[1]*u.x + f_in[3];
+	temp[4] = f_in[0]*uy2 - 2.0*f_in[2]*u.y + f_in[4];
+	temp[5] = f_in[0]*uxuy - f_in[1]*u.y - f_in[2]*u.x + f_in[5];	
+	//collision in central moments space
+	//calculate equilibrium distributions in cm space
+	cm_eq[0] = m00;
+	cm_eq[1] = -u.x*(m00 - 1.0);
+	cm_eq[2] = -u.y*(m00 - 1.0);
+	cm_eq[3] = m00*ux2 + 1./3.*m00 - ux2;
+	cm_eq[4] = m00*uy2 + 1./3.*m00 - uy2;
+	cm_eq[5] = uxuy*(m00 - 1.0);
+	cm_eq[6] = -u.y*(m00*ux2 + 1./3.*m00 - ux2 - 1./3.);
+	cm_eq[7] = -u.x*(m00*uy2 + 1./3.*m00 - uy2 - 1./3.);
+	cm_eq[8] = m00*ux2*uy2 + 1./3.*m00*ux2 + 1./3.*m00*uy2 + 1./9.*m00 - ux2*uy2 - 1./3.*ux2 - 1./3.*uy2;
+
+	//calculate forces in cm space
+	//He's scheme: get_continuous_Maxwellian_DF(dzeta=dzeta, psi=1, u=(ux, uy)) - velocity term
+	//collide eq: (eye(9)-S)*cm + S*cm_eq + (eye(9)-S/2.)*force_in_cm_space // SOI
+	f_in[0] = cm_eq[0];
+	f_in[1] = 0.5*Fhydro.x/rho + cm_eq[1];
+	f_in[2] = 0.5*Fhydro.y/rho + cm_eq[2];
+	f_in[3] = 0.5*cm_eq[3]*(s_b + s_v) + 0.5*cm_eq[4]*(s_b - s_v) - temp[3]*(0.5*s_b + 0.5*s_v - 1.0) - 0.5*temp[4]*(s_b - s_v);
+	f_in[4] = 0.5*cm_eq[3]*(s_b - s_v) + 0.5*cm_eq[4]*(s_b + s_v) - 0.5*temp[3]*(s_b - s_v) - temp[4]*(0.5*s_b + 0.5*s_v - 1.0);
+	f_in[5] = cm_eq[5]*s_v - temp[5]*(s_v - 1.0);
+	f_in[6] = 1./6.*Fhydro.y/rho + cm_eq[6];
+	f_in[7] = 1./6.*Fhydro.x/rho + cm_eq[7];
+	f_in[8] = cm_eq[8];
+
+	//back to raw moments
+	temp[0] = f_in[0];
+	temp[1] = f_in[0]*u.x + f_in[1];
+	temp[2] = f_in[0]*u.y + f_in[2];
+	temp[3] = f_in[0]*ux2 + 2.0*f_in[1]*u.x + f_in[3];
+	temp[4] = f_in[0]*uy2 + 2.0*f_in[2]*u.y + f_in[4];
+	temp[5] = f_in[0]*uxuy + f_in[1]*u.y + f_in[2]*u.x + f_in[5];
+	temp[6] = f_in[0]*ux2*u.y + 2.0*f_in[1]*uxuy + f_in[2]*ux2 + f_in[3]*u.y + 2.0*f_in[5]*u.x + f_in[6];
+	temp[7] = f_in[0]*u.x*uy2 + f_in[1]*uy2 + 2.0*f_in[2]*uxuy + f_in[4]*u.x + 2.0*f_in[5]*u.y + f_in[7];
+	temp[8] = f_in[0]*ux2*uy2 + 2.0*f_in[1]*u.x*uy2 + 2.0*f_in[2]*ux2*u.y + f_in[3]*uy2 + f_in[4]*ux2 + 4.0*f_in[5]*uxuy + 2.0*f_in[6]*u.y + 2.0*f_in[7]*u.x + f_in[8];
+	//back to density-probability functions
+	f_in[0] = temp[0] - temp[3] - temp[4] + temp[8];
+	f_in[1] = 0.5*temp[1] + 0.5*temp[3] - 0.5*temp[7] - 0.5*temp[8];
+	f_in[2] = 0.5*temp[2] + 0.5*temp[4] - 0.5*temp[6] - 0.5*temp[8];
+	f_in[3] = -0.5*temp[1] + 0.5*temp[3] + 0.5*temp[7] - 0.5*temp[8];
+	f_in[4] = -0.5*temp[2] + 0.5*temp[4] + 0.5*temp[6] - 0.5*temp[8];
+	f_in[5] = 0.25*temp[5] + 0.25*temp[6] + 0.25*temp[7] + 0.25*temp[8];
+	f_in[6] = -0.25*temp[5] + 0.25*temp[6] - 0.25*temp[7] + 0.25*temp[8];
+	f_in[7] = 0.25*temp[5] - 0.25*temp[6] - 0.25*temp[7] + 0.25*temp[8];
+	f_in[8] = -0.25*temp[5] - 0.25*temp[6] + 0.25*temp[7] + 0.25*temp[8];
+}
+
+CudaDeviceFunction void relax_and_collide_CM_phase_field(real_t f_in[9], real_t tau, vector_t u, vector_t F_phi, vector_t Fhydro, real_t rho) 
```

Comment:
`vector_t Fhydro` is useless here

Reply:
@ggruszczynski

Any opinion on the above?

Reply:
Naming:
- I will keep the `d2q9_pf_velocity` and override the old model.
- old mrt model was also in 'in press' but was placed in models/multiphase not models/article folder. I think that physics-related structure is clearer.

BC: Yes, I will add a switch.